### PR TITLE
fix some bugs of resolve-lock test & add config files for region schedule

### PR DESCRIPTION
**Pull request by youjiali1995**
scheduling

Signed-off-by: youjiali1995 <zlwgx1023@gmail.com>

### What problem does this PR solve? 

* Fix a concurrency bug of writing locks: the `mockLockTs` is not thread-safe.
* Add more logs for debugging.
* Add config files for cases involving region schedule.

### Check List 

Tests 
 - Manual test (add detailed scripts or steps below)

Code changes

 - Has Go code change

### Does this PR introduce a user-facing change?:

```release-note
NONE
```

### CODE REVIEW

File: argo/template/resolve-lock.yaml [hunk 1]
```diff
@@ -79,8 +73,5 @@ spec:
             -loki-addr={{inputs.parameters.loki-addr}} \
             -loki-username={{inputs.parameters.loki-username}} \
             -loki-password={{inputs.parameters.loki-password}} \
-            -matrix-config={{inputs.parameters.matrix-config}} \
-            -matrix-tidb={{inputs.parameters.matrix-tidb}} \
-            -matrix-tikv={{inputs.parameters.matrix-tikv}} \
-            -matrix-pd={{inputs.parameters.matrix-pd}} \
-            -matrix-sql={{inputs.parameters.matrix-sql}}
+            -tikv-config={{inputs.parameters.tikv-config}} \
+            -pd-config={{inputs.parameters.pd-config}} \
```
[the remark this answers is not in the document]
Comment (cosven) [reply argo/template/resolve-lock.yaml:77]: We can ignore this file.

**Review by cosven — APPROVED** [review]

File: cmd/resolve-lock/main.go [hunk 2]
```diff
@@ -31,8 +31,9 @@ import (
 var (
 	enableGreenGC = flag.Bool("enable-green-gc", true, "whether to enable green gc")
 	regionCount   = flag.Int("region-count", 200, "count of regions")
-	lockPerRegion = flag.Int("lock-per-region", 5, "count of locks in each region")
+	lockPerRegion = flag.Int("lock-per-region", 10, "count of locks in each region")
 	workers       = flag.Int("worker", 10, "count of workers to generate locks")
+	localMode     = flag.Bool("local-mode", false, "run the test in local mode which means use localhost cluster to run the test")
```
[the remark this answers is not in the document]
Comment (youjiali1995) [reply cmd/resolve-lock/main.go:36]: But the template of the status address is different `http://%s-tidb.%s.svc:10080`.

[the remark this answers is not in the document]
Comment (cosven) [reply cmd/resolve-lock/main.go:36]: We'd better use `node.IP:10080` to cook the tidb status address, and we need not to take care of the template.

File: tests/resolve-lock/resolve_lock.go [hunk 3]
```diff
@@ -83,6 +85,10 @@ func (c *resolveLockClient) openDB(ctx context.Context, ip string, port int32) e
 	if err != nil {
 		return errors.Trace(err)
 	}
+	_, err = db.ExecContext(ctx, fmt.Sprintf("DROP DATABASE `%s`", c.dbName))
```
Comment (zyguan) [inline tests/resolve-lock/resolve_lock.go:88]:
```suggestion
	_, err = db.ExecContext(ctx, fmt.Sprintf("DROP DATABASE IF EXISTS `%s`", c.dbName))
```

File: pkg/util/http/client.go [hunk 4]
```diff
@@ -26,14 +26,15 @@ func (c *Client) Get(url string) ([]byte, error) {
...
+
+	res, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
 		return nil, errors.Wrap(err, "Read GET response failed")
 	}
-	return body, nil
+	if resp.StatusCode != http.StatusOK {
+		return nil, errors.New(fmt.Sprintf("GET request \"%s\", got %v %s", url, resp.StatusCode, string(res)))
+	}
+	return res, nil
```
Comment (zyguan) [inline pkg/util/http/client.go:37]: What's the reason for checking status code after reading body?

Comment (youjiali1995) [reply pkg/util/http/client.go:37]: The body contains the reason why the status code is not 200.

File: cmd/resolve-lock/main.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (youjiali1995) [reply cmd/resolve-lock/main.go:36]: Do you mean I can replace `c.dbStatusAddr = fmt.Sprintf("http://%s-tidb.%s.svc:10080", dbNode.ClusterName, dbNode.Namespace)` by `fmt.Sprintf(http://%s:%d, dbNode.IP, dbNode.Port)`?

File: pkg/util/http/client.go (hunk 4 above)
Comment (zyguan) [reply pkg/util/http/client.go:37]: Ah, I see. I didn't notice that the `res` has been added to the error message.

File: cmd/resolve-lock/main.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (cosven) [reply cmd/resolve-lock/main.go:36]: Yes

**Pull request merged**